h2o-py/h2o/frame.py

```diff
@@ -1011,6 +1011,21 @@ def set_levels(self, levels):
         return H2OFrame._expr(expr=ExprNode("setDomain", self, False, levels), cache=self._ex._cache)
 
 
+    def rename(self, columns=None):
```

Comment:
What is the purpose of this method compared to `set_names`?

This method has one issue: suppose the original frame has columns: A, B and you want to rename B to A and A to B. The iterative approach won't work.

Reply:
See my reply to @jakubhava above.

Thanks for the feedback. Will fix in the next commit.

Reply:
I understand the purpose now, thanks.

I one way to fix the issue is remap the columns locally in this method and then call set_names on the full set of columns.